actionview/lib/action_view/cache_expiry.rb

```diff
@@ -2,9 +2,61 @@
 
 module ActionView
   class CacheExpiry
+    class ExecutionLock
```

Comment:
Wouldn't be possible to contribute this kind of lock into the `concurrent-ruby` itself? :thinking:

Reply:
I don't have a strong opinion about it, but I don't think it's enough of a general implementation to make sense to upstream to `concurrent-ruby`. That being said, we could ask? It's quite specific to the use case of the cache layer.